core/src/main/java/org/radargun/stats/RequestSet.java

```diff
@@ -0,0 +1,57 @@
+package org.radargun.stats;
+
+import org.radargun.Operation;
+
+/**
+ * @author Radim Vansa &lt;rvansa@redhat.com&gt;
+ */
+public final class RequestSet {
+   private final Statistics statistics;
+   private long begin = Long.MAX_VALUE;
+   private long end = Long.MIN_VALUE;
+   private long sumDurations = 0;
+   private boolean successful = true;
+
+   public RequestSet(Statistics statistics) {
+      this.statistics = statistics;
+   }
+
+   public void add(Request request) {
+      if (!request.isFinished()) {
+         throw new IllegalArgumentException();
+      }
+      begin = Math.min(begin, request.getRequestStartTime());
+      end = Math.max(end, request.getResponseCompleteTime());
+      sumDurations += request.duration();
+   }
+
+   public void finished(boolean successful, Operation operation) {
```

Comment:
What? It's just a convenience method when you get the result as boolean, so you pass that instead of doing an if.
However, now I've noticed that the assignment must happen before record call.